contracts/facets/ViewStateFacet.sol

```diff
@@ -260,9 +268,13 @@ contract ViewStateFacet {
             } else if (
                 !sarcophagus
                     .cursedArchaeologists[sarcophagus.cursedArchaeologistAddresses[i]]
-                    .isAccused
+                    .isAccused &&
+                    !sarcophagus.isCompromised &&
+                    !sarcophagus.isCleaned &&
+                    sarcophagus.resurrectionTime != 2 ** 256 - 1
```

Comment:
Definitely! Just stuck those in there as an afterthought when I realized that they were missing.

Is that out-of-gas scenario something that will apply to view functions being called by the webapp? It's true third parties could be trying to call it from their own contracts in the future as well but I'm most concerned about how it might impact us in the more immediate term.

Reply:
yes, out of gas can apply to view functions, but im not sure at what scale this would run out of gas.

Reply:
It sounds like you'd need to create a 163709 terabyte array to hit the out of gas limit but that execution time will suffer a lot sooner than that. We're safe for any reasonable array size here though and the user would run into problems creating a sarcophagus large enough to cause performance issues on this view function. Good thought though did not know view was also metered
https://betterprogramming.pub/issues-of-returning-arrays-of-dynamic-size-in-solidity-smart-contracts-dd1e54424235